kymatio/scattering2d/backend/torch_backend.py

```diff
@@ -0,0 +1,328 @@
+            NB : If direction is equal to 'C2R', then an error is raised.
+        
+        Raises
+        ------
+        RuntimeError
+            In the event that we are going from complex to real and not doing
+            the inverse fft or in the event x is not contiguous.
+        TypeError
+            In the event that x does not have a final dimension 2 i.e. not
+            complex. 
+        
+        Returns
+        -------
+        output : tensor
+            Result of FFT or IFFT.
+            
+    """
+    if direction == 'C2R':
+        if not inverse:
+            raise RuntimeError('C2R mode can only be done with an inverse FFT.')
+
+    if not iscomplex(x):
+        raise TypeError('The input should be complex (e.g. last dimension is 2).')
+
+    if not x.is_contiguous():
+        raise RuntimeError('Tensors must be contiguous!')
+
+    if direction == 'C2R':
+        output = torch.irfft(x, 2, normalized=False, onesided=False) * x.size(-2) * x.size(-3)
+    elif direction == 'C2C':
+        if inverse:
+            output = torch.ifft(x, 2, normalized=False) * x.size(-2) * x.size(-3)
+        else:
+            output = torch.fft(x, 2, normalized=False)
+
+    return output
+
+def cdgmm(A, B, inplace=False):
+    """Complex pointwise multiplication.
+    
+        Complex pointwise multiplication between (batched) tensor A and tensor B.
+
+        Parameters
+        ----------
+        A : tensor
+            A is a complex tensor of size (B, C, M, N, 2).
+        B : tensor
+            B is a complex tensor of size (M, N, 2) or real tensor of (M, N, 1).
+        inplace : boolean, optional
+            If set to True, all the operations are performed inplace.
+
+        Returns
+        -------
+        C : tensor
+            Output tensor of size (B, C, M, N, 2) such that:
+            C[b, c, m, n, :] = A[b, c, m, n, :] * B[m, n, :].
+
+    """
+    if not iscomplex(A):
+        raise TypeError('The input must be complex, indicated by a last '
+                        'dimension of size 2.')
+
+    if B.ndimension() != 3:
+        raise RuntimeError('The filter must be a 3-tensor, with a last '
+                           'dimension of size 1 or 2 to indicate it is real '
+                           'or complex, respectively.')
+
+    if not iscomplex(B) and not isreal(B):
+        raise TypeError('The filter must be complex or real, indicated by a '
+                        'last dimension of size 2 or 1, respectively.')
+
+    if A.size()[-3:-1] != B.size()[-3:-1]:
+        raise RuntimeError('The filters are not compatible for multiplication!')
+
+    if A.dtype is not B.dtype:
+        raise TypeError('A and B must be of the same dtype.')
+
+    if A.device.type != B.device.type:
+        raise TypeError('A and B must be both on GPU or both on CPU.')
+
+    if A.device.type == 'cuda':
+        if A.device.index != B.device.index:
+            raise TypeError('A and B must be on the same GPU!')
```

Comment:
Do we want the warnings that apply on the other backends?

Reply:
No - even if it would be better. A backend is responsible for its own low level resources.

Reply:
I don't understand this conversation. What warnings would that be?

Reply:
given the previous conversation, we should copy any "raise" in the code and add it to the `torch ` backend

Reply:
Oh, so for the docs? Sure, but we can do an overhaul of docs later.